lint/util.py

```diff
@@ -908,20 +908,30 @@ def find_windows_python(version):
     """Find the nearest version of python and return its path."""
 
     if version:
-        # On Windows, there may be no separately named python/python3 binaries,
-        # so it seems the only reliable way to check for a given version is to
-        # check the root drive for 'Python*' directories, and try to match the
+        from . import persist
+
+        # Try to find python.exe using py.exe
+        # Try the exact version first, then the major version
+        code = r'import sys;print(sys.executable)'
+        for version in (version, version.partition('.')[0]):
+            out = communicate(('py.exe', '-%s' % version), code)
+            if out:
+                path = out.splitlines()[0]
+                if ('python.exe' in path.lower()) and can_exec(path):
+                    persist.debug('find_windows_python: <=', path)
+                    return path
+
+        # Check the root drive for 'Python*' directories, and try to match the
         # version based on the directory names. The 'Python*' directories end
         # with the <major><minor> version number, so for matching with the version
         # passed in, strip any decimal points.
         stripped_version = version.replace('.', '')
         prefix = os.path.abspath(os.path.join(
-            os.environ.get("SYSTEMROOT", "\\")[:2],
+            os.environ.get("SYSTEMDRIVE", "\\"),
```

Comment:
This change has the same problem the code already had. Please see #490.
I like your change better than mine but you should adjust it to fix the problem I've fixed (append backslash).
The code could just say:
`os.path.join(os.environ.get('SYSTEMDRIVE', ''), '\\', 'Python')`